### PR TITLE
Melee Attacks Closest Ally

### PR DESCRIPTION
**Pull request by Anisetobegan**

### CODE REVIEW

File: Assets/Scripts/Character.cs [hunk 1]
```diff
@@ -39,6 +39,8 @@ enum State
     Vector2 startPos;
     Vector2 endPos;
 
+    public CapsuleCollider playerCollider;
+
```
Comment (morathias) [inline Assets/Scripts/Character.cs:45]: Make this private Serialized field and make a getter

File: Assets/Scripts/GameManager.cs [hunk 2]
```diff
@@ -16,6 +16,13 @@ public class GameManager : MonoBehaviour
 
     [SerializeField] private GameObject waveScreen;
 
+    public Transform PlayerTransform
+    {
+        get { return player.transform; }
+    }
+
+    public float playerColliderRadius { get { return player.playerCollider.radius; } }
```
Comment (morathias) [inline Assets/Scripts/GameManager.cs]: Use getter here

File: Assets/Scripts/MeleeScript.cs [hunk 3]
```diff
@@ -2,10 +2,15 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.Scripting.APIUpdating;
+using UnityEngine.AI;
+using Unity.VisualScripting;
```
Comment (morathias) [inline Assets/Scripts/MeleeScript.cs]: Remove VisualScripting

File: Assets/Scripts/MeleeScript.cs [hunk 4]
```diff
@@ -15,13 +20,101 @@ enum State
...
 
+    protected override void Move()
+    {
+        Vector3 offset = target + (transform.position - target).normalized * (GameManager.Instance.playerColliderRadius + 1f);
+        agent.SetDestination(offset);
     }
 
     protected override void Attack()
     {
+        Debug.Log("Attacking");
```
Comment (morathias) [inline Assets/Scripts/MeleeScript.cs]: Remove Debug

File: Assets/Scripts/MeleeScript.cs [hunk 5]
```diff
@@ -15,13 +20,101 @@ enum State
...
+
+        agent.SetDestination(transform.position);
```
Comment (morathias) [inline Assets/Scripts/MeleeScript.cs]: Use agent.Stop() instead of pasing it's position

**Review by morathias — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Character.cs (hunk 1 above)
Comment (Anisetobegan) [reply Assets/Scripts/Character.cs:45]: Done

File: Assets/Scripts/GameManager.cs (hunk 2 above)
Comment (Anisetobegan) [reply Assets/Scripts/GameManager.cs]: Done

File: Assets/Scripts/MeleeScript.cs (hunk 3 above)
Comment (Anisetobegan) [reply Assets/Scripts/MeleeScript.cs]: Done

File: Assets/Scripts/MeleeScript.cs (hunk 4 above)
Comment (Anisetobegan) [reply Assets/Scripts/MeleeScript.cs]: Done

File: Assets/Scripts/MeleeScript.cs (hunk 5 above)
Comment (Anisetobegan) [reply Assets/Scripts/MeleeScript.cs]: Done

**Review by morathias — APPROVED** [review]

**Pull request merged**